From 1524b95226959c38c3f1362aa8fb387b9aaf1d8d Mon Sep 17 00:00:00 2001
From: Serguei Katkov <serguei.i.katkov@intel.com>
Date: Wed, 7 Aug 2013 01:11:48 +0700
Subject: Dalvik: Fix similarInstructions utility

BZ: 128555

similarInstructions is too smart. In case some of use comes from out of trace
we will not be able to prove that two instructions generates the same value.
So similarInstructions should report true (possible the same) result.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I8ffb1c079379640cf3be70ba3461b0032f365268
Orig-MCG-Change-Id: I29758b73cac57034b44fd867e7879bf071860765
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Reviewed-on: http://android.intel.com:8080/125084
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/LoopRegisterUsage.cpp |   12 +-----------
 1 files changed, 1 insertions(+), 11 deletions(-)

diff --git a/vm/compiler/LoopRegisterUsage.cpp b/vm/compiler/LoopRegisterUsage.cpp
index 6a6193e..ae63ffb 100644
--- a/vm/compiler/LoopRegisterUsage.cpp
+++ b/vm/compiler/LoopRegisterUsage.cpp
@@ -473,17 +473,7 @@ static bool similarInstructions (MIR *first, MIR *second, const std::map<MIR *,
         //It is possible the use is not defined in the trace
         if (defFirst == 0 || defSecond == 0)
         {
-            //If one is defined but not the other, then who knows if they alias actually
-            if (defFirst != 0 || defSecond != 0)
-            {
-                return true;
-            }
-
-            //If this happens, then the use actually must be the same SSA register
-            if (ssaRepFirst->uses[i + startUseIndexFirst] != ssaRepSecond->uses[i + startUseIndexSecond])
-            {
-                return false;
-            }
+            return true;
         }
         else
         {
-- 
1.7.4.1

